packages/cdktf/lib/terraform-remote-state.ts

```diff
@@ -0,0 +1,64 @@
+import { Construct } from "constructs";
+import { TerraformElement } from "./terraform-element";
+import { Token } from "./tokens";
+import { deepMerge } from "./util";
+
+export interface TerraformRemoteStateConfig {
+    readonly workspace?: string;
+    readonly defaults?: { [key: string]: any };
+}
```

Comment:
Does this apply to all remote state data sources?

Reply:
Yeah, they do. Looks like I missed that in the renaming pass, though I'm not sure what the best name would be.

Some other approaches:

- Change to be closer to how `TerraformMetaArguments` and `TerraformResourceConfig` are setup and include as a base interface
- Make them optional parameters on the constructor
- Just have them be properties

Thoughts:

- Current approach is a bit awkward to use.
- The meta route is nice except that the config is shared with backends.
- Not all potential languages support optional parameters, so that isn't great.
- Having to create a local variable to set is inconvenient.

Reply:
> * Change to be closer to how `TerraformMetaArguments` and `TerraformResourceConfig` are setup and include as a base interface

That's what I thought as well 👍 

> * The meta route is nice except that the config is shared with backends.

Couldn't the remote state props extend from `TerraformRemoteStateConfig` and `S3BackendProps`?